### PR TITLE
Upstream sync

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Keep in sync with upstream

## What

Follop up #448
Upstream sync 2b42f4005c3a2c30c8bd39da3e167ac19a3a28b7..df92191e4d8e7a5ff57e7d4bed79a1a2a1758684

## Tests

CI

### CODE REVIEW

File: tracer/src/Datadog.Trace.ClrProfiler.Native/always_on_profiler.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include <algorithm>
 #ifndef _WIN32
   #include <pthread.h>
+  #include<codecvt>
```
Comment (pellared) [inline tracer/src/Datadog.Trace.ClrProfiler.Native/always_on_profiler.cpp:10]:
```suggestion
  #include <codecvt>
```

File: profiler/test/Datadog.Profiler.Native.Tests/ConfigurationTest.cpp [hunk 2]
```diff
@@ -153,9 +153,9 @@ TEST(ConfigurationTest, CheckDefaultPprofDirectoryWhenVariableIsNotSet)
     auto configuration = Configuration{};
     auto expectedValue =
 #ifdef _WINDOWS
-        WStr("C:\\ProgramData\\Datadog-APM\\Pprof-files\\DotNet");
+        WStr("C:\\ProgramData\\SignalFx .NET Tracing\Pprof-files\\DotNet");
```
Comment (pellared) [inline profiler/test/Datadog.Profiler.Native.Tests/ConfigurationTest.cpp:156]:
```suggestion
        WStr("C:\\ProgramData\\SignalFx .NET Tracing\\Pprof-files\\DotNet");
```

also this files misses the "Modified by Splunk" comment

**Review by theletterf — APPROVED** [review]

File: profiler/test/Datadog.Profiler.Native.Tests/ConfigurationTest.cpp (hunk 2 above)
Comment (Kielek) [reply profiler/test/Datadog.Profiler.Native.Tests/ConfigurationTest.cpp:156]: As all/most others cpp files ;)

**Review by pellared — APPROVED** [review]
I am exhausted 😄

**Pull request merged**